src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java

```diff
@@ -65,10 +68,77 @@ public PartAlignmentOffset findOffsets(Part part, BoardLocation boardLocation,
 
         // Pre-rotate to minimize runout
         double preRotateAngle = 0;
-        if (preRotate && boardLocation != null && placementLocation != null) {
-            preRotateAngle =
-                    Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation)
-                           .getRotation();
+        if (preRotate) {
+			if(part == null || nozzle.getPart() == null) 					   { throw new Exception("no part on nozzle"); }			
+			if((!part.toString().equals(nozzle.getPart().toString()))) { throw new Exception("Part mismatch with part on nozzle"); }
+			part=nozzle.getPart();
+			double angle = placementLocation.getRotation();
+			if(boardLocation!=null) {
+				angle = Utils2D.calculateBoardPlacementLocation(boardLocation, placementLocation).getRotation();              
+			}
+			while (Math.abs(angle) > 180.0) { angle += (angle < 0.)? 360. : -360.; } 
+			double nozzleAngle=angle;
+			double placementAngle=angle;
+            MovableUtils.moveToLocationAtSafeZ(nozzle, VisionUtils.getBottomVisionCamera().getLocation().add(new Location(part.getHeight().getUnits(), 0.0, 0.0,part.getHeight().getValue(), 0.0)).derive(null, null, null, angle));
+            CvPipeline pipeline = partSettings.getPipeline();
+            pipeline.setCamera(VisionUtils.getBottomVisionCamera());
+            pipeline.setNozzle(nozzle);
+			pipeline.setValue(angle);
```

Comment:
setValue is double type variable that just set a value.
In this case the value variable inside pipeline holds the target part angle value, that differ from
nozzle rotation value.
It is used as example inside ImageCapture when part size is bigger as camera FOV or inside MaskRectangle when value is less then 3 (double) where it set the rectangle to part size,
Further when having prerotate without having access to this variable it is not possible to extract
footprints like this:
https://cloud.githubusercontent.com/assets/9506454/25804378/d56d8682-33fb-11e7-8c2a-92b63ef54c6d.png
and as consequence drop the ic because of coplanarity error .